android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateUtils.java

```diff
@@ -176,11 +180,13 @@ public static void verifyFolderHash(String folderPath, String expectedHash) {
 
     public static Map<String, Object> verifyAndDecodeJWT(String jwt, PublicKey publicKey) {
         try {
-            final JWTVerifier verifier = new JWTVerifier(publicKey);
-            final Map<String, Object> claims = verifier.verify(jwt);
-            CodePushUtils.log("JWT verification succeeded:\n" + claims.toString());
-            return claims;
-        } catch (Exception e) {
+            SignedJWT signedJWT = SignedJWT.parse(jwt);
+            JWSVerifier verifier = new RSASSAVerifier((RSAPublicKey)publicKey);
+            if (signedJWT.verify(verifier)) {
+                return signedJWT.getJWTClaimsSet().getClaims();
```

Comment:
Does it make sense to keep logging here to help devs to debug going further?

Reply:
fixed, thanks!